benchmark/single-source/InsertCharacter.swift

```diff
@@ -0,0 +1,115 @@
+//===--- InsertCharacter.swift ------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import TestsUtils
+
+public let InsertCharacter = [
+    BenchmarkInfo(name: "InsertCharacterEndIndex", runFunction: run_InsertCharacterEndIndex, tags: [.validation, .api], setUpFunction: buildWorkload),
+    BenchmarkInfo(name: "InsertCharacterTowardsEndIndex", runFunction: run_InsertCharacterTowardsEndIndex, tags: [.validation, .api], setUpFunction: buildWorkload),
+    BenchmarkInfo(name: "InsertCharacterStartIndex", runFunction: run_InsertCharacterStartIndex, tags: [.validation, .api], setUpFunction: buildWorkload),
+    BenchmarkInfo(name: "InsertCharacterEndIndexNonASCII", runFunction: run_InsertCharacterEndIndexNonASCII, tags: [.validation, .api], setUpFunction: buildWorkload),
+    BenchmarkInfo(name: "InsertCharacterTowardsEndIndexNonASCII", runFunction: run_InsertCharacterTowardsEndIndexNonASCII, tags: [.validation, .api], setUpFunction: buildWorkload),
+    BenchmarkInfo(name: "InsertCharacterStartIndexNonASCII", runFunction: run_InsertCharacterStartIndexNonASCII, tags: [.validation, .api], setUpFunction: buildWorkload)
+]
+
+let str = String(repeating: "A very long ASCII string.", count: 200)
+
+func buildWorkload() {
+    blackHole(str)
+}
+
+@inline(never)
+func run_InsertCharacterEndIndex(_ N: Int) {
+    var workload = str
+    let character: Character = "a"
+    let scale = 3000
+    for _ in 0..<N * scale {
+        workload.insert(character, at: workload.endIndex)
+    }
+    blackHole(workload)
+    CheckResults(workload.count == str.count + N * scale)
```

Comment:
I'm a little worried that this check could dominate the benchmark, as `count` is linear on the very String we're building up, and might be a more sophisticated operation than even just append. This may just not be a check we can do.

Reply:
Do you have more details on how to use the profiler to gain more insights on how much `CheckResults` takes?

Reply:
The easiest way, if you have this extracted into a stand-alone file, you can run something like:

`swiftc -O my_benchmark.swift && instruments -t "Time Profiler" ./my_benchmark`. Then check where time is being spent for each of these benchmark functions. You'll have to read a little into things since the results are post-inlining, but basically it's a sanity check that the benchmark is spending time in the thing you're trying to benchmark, i.e. `insert`.

If you don't have a standalone file, you can still instrument the `Benchmark_O` binary, passing the specific benchmark you want as an argument on the command line.

Reply:
The `count` operation takes ~2-3% of the time in the benchmark after profiling it a couple of times. I'm removing all the `CheckResults` to be sure most of the time is spent in inserting rather than asserting the actual operation.